modules/logging/filewriter_test.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2015 Matthew Holt and The Caddy Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"os"
+	"path"
+	"syscall"
+	"testing"
+
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+)
+
+func TestFileCreationMode(t *testing.T) {
+	on := true
+	off := false
+
+	tests := []struct {
+		name     string
+		fw       FileWriter
+		wantMode os.FileMode
+	}{
+		{
+			name: "default mode no roll",
+			fw: FileWriter{
+				Roll: &off,
+			},
+			wantMode: 0o600,
+		},
+		{
+			name: "default mode roll",
+			fw: FileWriter{
+				Roll: &on,
+			},
+			wantMode: 0o600,
+		},
+		{
+			name: "custom mode no roll",
+			fw: FileWriter{
+				Roll: &off,
+				Mode: 0o666,
+			},
+			wantMode: 0o666,
+		},
+		{
+			name: "custom mode roll",
+			fw: FileWriter{
+				Roll: &on,
+				Mode: 0o666,
+			},
+			wantMode: 0o666,
+		},
+	}
+
+	m := syscall.Umask(0o000)
```

Comment:
Tests are failing on windows due to the umask syscall not defined (God bless CI/CD !).
What would your recommendation be to manage this : having a helper mock that does nothing on windows ? That would imply having additional filewriter_test_unix.go and filewriter_test_windows.go  files to define the mock ?

Reply:
Yeah probably. Make 2 files, one for windows and one for unix (I guess) which both define the same function, but make the Windows one do nothing.

Reply:
Thanks @francislavoie. I submitted a new commit hoping it fixes the issue (I have no access to a windows machine to test). Let me know if you prefer an exhaustive 

```go
//go:build darwin || dragonfly || freebsd || linux || nacl || netbsd || openbsd || solaris
```

or a simpler 

```go
//go:build !windows
```